### PR TITLE
Fix regression loading large files (>2GB) on 32bit systems ##io

### PR DESCRIPTION
**Pull request by radare**
This PR does nothing, just aesthetics for now

 

**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md#code-style)
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [radare2 book](https://github.com/radareorg/radare2book) with the relevant information (if needed)

**Detailed description**

...

**Test plan**

...

**Closing issues**

...

### CODE REVIEW

**Comment by ret2libc** [comment]
this is super WIP, right? :)

**Comment by radare** [comment]
Yes, no need to review this pr , im waiting for feedback from the friend who foundnu this issue i didnt had time to review/check but the issue is there, and after reviewing the code i found no issues, but just few aesthetical changes. Ignore this pr until i remove the WIP bit

**Comment by radare** [comment]
Ref https://github.com/radareorg/radare2/issues/16590

**Comment by radare** [comment]
i think the problem is a regression. and should be fixed by bring back this https://users.suse.com/~aj/linux_lfs.html

**Comment by radare** [comment]
this is imho no longer a draft, this PR should be good for testing and then merge when fix is confirmed

**Comment by radare** [comment]
should be fixed now,, wonder when this regression got introduced. maybe after siol?

File: libr/include/r_io.h [hunk 1]
```diff
@@ -3,6 +3,9 @@
 #ifndef R2_IO_H
 #define R2_IO_H
 
+#define _GNU_SOURCE 1
```
Comment (ret2libc) [inline libr/include/r_io.h:6]: This produces a lot of warnings while compiling, see https://github.com/radareorg/radare2/pull/16648/checks?check_run_id=622122422#step:7:364

**Comment by Shaddy** [comment]
The only way got this working was defining it at `libr/util/sandbox.c`

`#define _FILE_OFFSET_BITS 64`

Unfortunately seems that io_default.c/r_io.h does not apply for sandbox.c as it is not using those headers, and open seems is not getting the definition.

File: libr/include/r_io.h (hunk 1 above)
Comment (ret2libc) [reply libr/include/r_io.h:6]: This is still no good, it produces a lot of warnings.
Maybe do:
```C
#ifndef _GNU_SOURCE
#define _GNU_SOURCE 1
#endif
```

**Comment by ret2libc** [comment]
@radare just make sure _GNU_SOURCE is not creating 123231 warnings, then since the issue is fixed according to https://github.com/radareorg/radare2/issues/16590#issuecomment-620262094 , we can probably merge.

**Comment by Shaddy** [comment]
@radare I've tested it and works perfect

**Comment by radare** [comment]
YAY

**Pull request merged**